flair/file_utils.py

```diff
@@ -82,9 +82,12 @@ def cached_path(url_or_filename: str, cache_dir: Union[str, Path]) -> Path:
     return the path to the cached file. If it's already a local path,
     make sure the file exists and then return the path.
     """
-    if type(cache_dir) is str:
-        cache_dir = Path(cache_dir)
-    dataset_cache = flair.cache_root / cache_dir
+    cache_dir = Path(cache_dir)
+
+    if flair.cache_root not in cache_dir.parents:
```

Comment:
what does this check do?

Reply:
This checks if the cache_dir is already part of the cache_root and doesn't duplicate the path. E.g.: makes the cached files being placed in `cache/flair/...` instead of `cache/flair/cache/flair/...`.